### DISCUSSION TITLE
How do I build from the command line without ever loading the Visual Studio GUI?

### DISCUSSION THREAD

**Discussion by dulsi** [microsoft/vcpkg#20235] [Q&A] [answered]
If I grab our source tree and try to build from the command line, it fails saying it can't find some cppunit header files. If I load the solution in Visual Studio, I can build from there. After doing that I can build from the command line without a problem. What is the GUI doing to enable this?

**Comment by kossiLeBeau** [comment]
Go to the **compiler command prompt.** 
Change to the folder containing vcpkg.exe
Issue the command: vcpkg install <package Name>

**Reply by dulsi** [reply]
No. You misunderstand the problem. We did:

.\vcpkg\bootstrap-vcpkg.bat
.\vcpkg\vcpkg integrate install
\vcpkg\vcpkg install cppunit

Afterward the program fails to build from the command line. Rebooting doesn't matter.

If I then load Visual Studio project and compile in there it works. If I clean the solution and rebuild from the command line, it works. Therefore something changed when I loaded and compiled with Visual Studio. (It might be just loading Visual Studio. I haven't tested that.) It isn't in the solution files, because if I download a clean copy of the source to a different directory, it builds fine.

**Comment by dulsi** [comment, marked as answer]
I think we found our issue. I had thought that you needed to run Visual Studio GUI to get it to recognize vcpkg. It appears I must have tried to build the 32 bit version of the solution and since the 32 bit version of the library wasn't installed it didn't work.

The problem with Jenkins seems to be that it alters the LocalAppData setting. By running vcpkg integrate install before building in Jenkins it setups the %LocalAppData%\vcpkg directory and everything runs.
